plugins/org.orbisgis.groovyeditor/test/org/orbisgis/ui/editors/groovy/ClassPathHandlerTest.java

```diff
@@ -0,0 +1,82 @@
+package org.orbisgis.ui.editors.groovy;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ClassPathHandlerTest {
+
+    private List<URL> urls;
+    ClassPathHandler classPathHandler;
+
+    @BeforeEach
+    void beforeEach(){
+        urls = new ArrayList<URL>();
+        classPathHandler = new ClassPathHandler();
+    }
+
+    @AfterEach
+    void afterEach() throws MalformedURLException {
+        if (classPathHandler.getUrls().size() > 0){
+            for(int i=0; i< classPathHandler.getUrls().size(); i++){
+                classPathHandler.deleteOneURL(i);
+            }
+        }
+    }
+
+    @Test
+    void addPathToUrlListIfDirectoryTest() throws MalformedURLException {
+        urls.add(new File("///d:" + "/notAJar").toURI().toURL());
```

Comment:
Your URL list (I assume that it is the expected list of urls) should be only in the method scope and not in a global scope.